components/zcash_address/src/kind/unified.rs

```diff
@@ -132,52 +132,59 @@ pub(crate) mod private {
     use zcash_encoding::CompactSize;
 
     /// A raw address or viewing key.
-    pub trait SealedReceiver:
+    pub trait SealedItem:
         for<'a> TryFrom<(u32, &'a [u8]), Error = ParseError> + cmp::Ord + cmp::PartialOrd + Clone
     {
         fn typecode(&self) -> Typecode;
         fn data(&self) -> &[u8];
     }
 
     /// A Unified Container containing addresses or viewing keys.
-    pub trait SealedContainer: super::ToReceivers {
+    pub trait SealedContainer: super::Container {
         const MAINNET: &'static str;
         const TESTNET: &'static str;
         const REGTEST: &'static str;
 
-        fn from_inner(receivers: Vec<Self::Receiver>) -> Self;
+        /// Implementations of this method should act as unchecked constructors
+        /// of the container type; the caller is guaranteed to check the
+        /// general invariants that apply to all unified containers.
+        fn from_inner(items: Vec<Self::Item>) -> Self;
 
-        /// Returns the raw encoding of this Unified Address or viewing key.
-        fn to_bytes(&self, hrp: &str) -> Vec<u8> {
-            assert!(hrp.len() <= PADDING_LEN);
-
-            let mut writer = std::io::Cursor::new(Vec::new());
-            for receiver in &self.receivers() {
-                let addr = receiver.data();
+        fn write_raw_encoding<W: Write>(&self, mut writer: W) {
+            for item in &self.items() {
+                let addr = item.data();
                 CompactSize::write(
                     &mut writer,
-                    <u32>::from(receiver.typecode()).try_into().unwrap(),
+                    <u32>::from(item.typecode()).try_into().unwrap(),
                 )
                 .unwrap();
                 CompactSize::write(&mut writer, addr.len()).unwrap();
                 writer.write_all(addr).unwrap();
             }
+        }
+
+        /// Returns the jumbled padded raw encoding of this Unified Address or viewing key.
+        fn to_bytes(&self, hrp: &str) -> Vec<u8> {
```

Comment:
This shouldn't be named `to_bytes`, because that is what a public method that output the raw bytes would use.
```suggestion
        /// Returns the jumbled padded raw encoding of this Unified Address or viewing key,
        /// as used in the string encoding.
        fn to_jumbled_bytes(&self, hrp: &str) -> Vec<u8> {
```
Alternatively, if this method is deleted by #465 (which could inline its contents into an encoding function) then I'd be fine with leaving this as-is in this PR.

Reply:
This will be fixed in #465